src/SFA.DAS.Approvals.Api/Controllers/ProvidersController.cs

```diff
@@ -43,5 +43,21 @@ public async Task<IActionResult> GetAll()
                 return BadRequest();
             }
         }
+
+        [HttpGet]
+        [Route("{providerId}/courses/{trainingCode}")]
+        public async Task<IActionResult> GetProviderCoursesDeliveryModel(long providerId, string trainingCode)
+        {
+            try
+            {
+                var result = await _mediator.Send(new GetProviderCoursesDeliveryModelQuery(providerId, trainingCode));
+                return Ok(result);
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e, "Error getting Provider Courses Delivery Models for Provider {providerId} and course {trainingCode}", providerId, trainingCode);
+                return BadRequest();
```

Comment:
This seems more like an `InternalServerError` than a bad request don't you think?

A bad request would be if the provider ID was less than 1, or the training code was blank.